core/src/main/scala/com/databricks/labs/remorph/parsers/intermediate/extensions.scala

```diff
@@ -2,4 +2,20 @@ package com.databricks.labs.remorph.parsers.intermediate
 
 trait AstExtension
 
-case class Column(name: String) extends Expression with AstExtension {}
\ No newline at end of file
+case class Column(name: String) extends Expression with AstExtension {}
+
+abstract class Unary(pred: Expression) extends Expression {}
+abstract class Binary(left: Expression, right: Expression) extends Expression {}
+
+trait Predicate extends Expression with AstExtension
+
+case class And(left: Predicate, right: Predicate) extends Binary(left, right) with Predicate {}
+case class Or(left: Predicate, right: Predicate) extends Binary(left, right) with Predicate {}
+case class Not(pred: Predicate) extends Unary(pred) with Predicate {}
```

Comment:
```suggestion
case class And(left: Expression, right: Expression) extends Binary(left, right) with Predicate {}
case class Or(left: Expression, right: Expression) extends Binary(left, right) with Predicate {}
case class Not(pred: Expression) extends Unary(pred) with Predicate {}
```
otherwise this will break on `1 OR 2`, `IS_MEMBER(x) AND group_name = x`, `NOT enabled`, etc, etc. we're dealing with (unresolved) logical plan here, not a concrete execution plan.

Reply:
and the deciding factor is:
```
case class EqualTo(left: Expression, right: Expression)
    extends BinaryComparison with NullIntolerant

...

abstract class BinaryComparison extends BinaryOperator with Predicate
```